canarytokens/canarydrop.py

```diff
@@ -244,39 +247,27 @@ class Config:
             datetime: lambda v: v.strftime("%s.%f"),
         }
 
-    def add_additional_info_to_hit(
-        self,
-        hit_time: str,
-        additional_info: dict[str, str],
-    ) -> None:
-        """ """
-        trigger_details = queries.get_canarydrop_triggered_details(self.canarytoken)
-        if not any([hit_time == o.time_of_hit for o in trigger_details.hits]):
-            raise ValueError(
-                textwrap.dedent(
-                    """
-                Got additional details for a hit that does not exist.
-                This is likely a use case we don't support yet but can.
-            """
-                )
-            )
-
-        queries.add_additional_info_to_hit(self.canarytoken, hit_time, additional_info)
-
     def add_canarydrop_hit(self, *, token_hit: AnyTokenHit):
         """Adds a hit to the drops history `.triggered_details`.
 
         Args:
             token_hit (AnyTokenHit): Hit to add.
         """
-        queries.save_canarydrop(self)
+        if self.triggered_details.token_type != token_hit.token_type:
+            # Design: This might not hold in the future but for now this is true.
+            raise ValueError(
+                f"All hits must be of a single type. Given {token_hit.token_type}; existing {self.triggered_details.token_type}"
+            )
 
         self.triggered_details.hits.append(token_hit)
+        if len(self.triggered_details.hits) > switchboard_settings.MAX_HISTORY:
+            self.triggered_details.hits = self.triggered_details.hits[
+                -min(
+                    len(self.triggered_details.hits), switchboard_settings.MAX_HISTORY
+                ) :  # noqa: E203
+            ]
```

Comment:
I think this would be way more readable if you broke out this index as a local variable. Something like
```suggestion
            max_hits = min(len(self.triggered_details.hits), switchboard_settings.MAX_HISTORY)
            self.trigger_details.hits = self.triggered_details.hits[-max_hits:]
```